### DISCUSSION TITLE
Automatic Changelog

### DISCUSSION THREAD

**Discussion by loicguilmard** [PyMoDAQ/PyMoDAQ#477] [Ideas]
We talked about it during the pymodays.

what do you think about just getting the list from the release description , like :

# Changelog

### What's Changed
* Improve the serializer by @BenediktBurger in https://github.com/PyMoDAQ/PyMoDAQ/pull/430
* Corrected bug in remote_manager by @rbeignon in https://github.com/PyMoDAQ/PyMoDAQ/pull/438
* Update GitHub Action Versions by @seb5g in https://github.com/PyMoDAQ/PyMoDAQ/pull/437

### New Contributors
* @rbeignon made their first contribution in https://github.com/PyMoDAQ/PyMoDAQ/pull/438

**Full Changelog**: https://github.com/PyMoDAQ/PyMoDAQ/compare/4.4.6...4.4.7
### What's Changed
* Add long_description_content_type to setup by @rgeneaux in https://github.com/PyMoDAQ/PyMoDAQ/pull/417
* Add example for qt-less standalone module using LECO by @BenediktBurger in https://github.com/PyMoDAQ/PyMoDAQ/pull/413
* Features/binary transport for PyLECO by @BenediktBurger in https://github.com/PyMoDAQ/PyMoDAQ/pull/404
* use base_units to get the displayed unit by @seb5g in https://github.com/PyMoDAQ/PyMoDAQ/pull/425
* get new current value when user change the actuator axis by @seb5g in https://github.com/PyMoDAQ/PyMoDAQ/pull/426
* add specific condition for displaying correct units by @seb5g in https://github.com/PyMoDAQ/PyMoDAQ/pull/427

**Full Changelog**: https://github.com/PyMoDAQ/PyMoDAQ/compare/4.4.5...4.4.6
## [4.4.5 Bayesian Optimisation compat](https://github.com/PyMoDAQ/PyMoDAQ/releases/tag/4.4.5)
**Full Changelog**: https://github.com/PyMoDAQ/PyMoDAQ/compare/4.4.4...4.4.5

##409 shows a major breaking chnage in the Bayesian Optimisation package. Until this sis included in pymodaq, we will restrict this libray to the <2.0.0 version
## [4.4.4](https://github.com/PyMoDAQ/PyMoDAQ/releases/tag/4.4.4)
**Full Changelog**: https://github.com/PyMoDAQ/PyMoDAQ/compare/4.4.3...4.4.4

correct units printing in DAQ_Move (reduced units wr to base units) See pint documention
Special case of angular degrees
### What's Changed
* Update move_utility_classes.py by @seb5g in https://github.com/PyMoDAQ/PyMoDAQ/pull/398
* axis, units initialization and reading from client by @seb5g in https://github.com/PyMoDAQ/PyMoDAQ/pull/400

**Full Changelog**: https://github.com/PyMoDAQ/PyMoDAQ/compare/4.4.2...4.4.3
## [4.4.2](https://github.com/PyMoDAQ/PyMoDAQ/releases/tag/4.4.2)
**Full Changelog**: https://github.com/PyMoDAQ/PyMoDAQ/compare/4.4.1...4.4.2
### What's Changed
* Add updater.yml for checking for GitHub actions updates by @runrunm in https://github.com/PyMoDAQ/PyMoDAQ/pull/388
* Update GitHub Action Versions by @seb5g in https://github.com/PyMoDAQ/PyMoDAQ/pull/390
* Doc/cleaning reordering by @seb5g in https://github.com/PyMoDAQ/PyMoDAQ/pull/393
* patched consistent behaviour for multi axes by @seb5g in https://github.com/PyMoDAQ/PyMoDAQ/pull/394

**Comment by BenediktBurger** [comment]
Depends on the quality of the descriptions.

If there is no manual adjustement afterwards, it is important to change the PR titles/descriptions before merging.

**Comment by loicguilmard** [comment]
My idea was to push a Changelog.md automatically.
We can always edit it afterwards or change the generation but we should be able to automate it.

For instance, what would be a good automatic PR title/description ?

On the other hand, from the disccussions i could read about what is a changelog. Most of the philosophical answers were that a changelog is not a release list. :)  (I really like [Ned Batchelder](https://nedbatchelder.com/) :D )
Therefore, the courageous maintainers are writing it themselves as a separate summary of what is really new/important in a release, but it's not what was discussed at the pymodayz.
May be we should have two files ? AutomaticChangelog.md and Changelog.md that is edited manually ? The automatic one would just prepare the "real" changelog ?

**Comment by loicguilmard** [comment]
@seb5g @malik-irain any thoughts ?

**Comment by malik-irain** [comment]
I think it should be automated, even if it's only local, at least for the ease of seeing everything since the last release, which is especially useful if someone were to manually edit the changelog. 

As for the two versions, it seems like the automated one would give too much useless details like automatically generated PR such as "Update GitHub Action Versions". I don't think it's useful to show it, when the link to the full changelog is provided anyways.

In all the cases, as Benedikt said, we should make an effort to give good PR names (not always so easy unfortunately...)

**Comment by seb5g** [comment]
I would actually prefer a manual one that translates the changes from the PR to some human readable summary. It's another task but could allow people to follow evolutions
